mediapipe/graphs/instantmotiontracking/subgraphs/calculators/tracked_anchor_manager_calculator.cc

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mediapipe/framework/calculator_framework.h"
+#include "mediapipe/framework/formats/detection.pb.h"
+#include "mediapipe/framework/formats/location_data.pb.h"
+#include "mediapipe/framework/port/status.h"
+#include "mediapipe/framework/port/ret_check.h"
+#include "mediapipe/util/tracking/box_tracker.h"
+#include "mediapipe/graphs/instantmotiontracking/calculators/transformations.h"
+
+namespace mediapipe {
+
+constexpr char kAnchorsTag[] = "ANCHORS";
+constexpr char kBoxesInputTag[] = "BOXES";
+constexpr char kBoxesOutputTag[] = "START_POS";
+constexpr char kCancelTag[] = "CANCEL_ID";
+
+// Generates the final model matrix rotations via the IMU (orientation sensors)
+// of the device and user rotation information
+//
+// Input:
+//  ANCHORS - Initial anchor data (tracks changes and where to re/position)
+//  BOXES - Used in cycle, boxes being tracked meant to update positions
+// Output:
+//  START_POS - Positions of boxes being tracked (can be overwritten with ID)
+//  CANCEL_ID - Single integer ID of tracking box to remove from tracker subgraph
+//  ANCHORS - Updated set of anchors with tracked and normalized X,Y,Z
+//
+// Example config:
+// node {
+//   calculator: "TrackedAnchorManagerCalculator"
+//   input_stream: "ANCHORS:initial_anchor_data"
+//   input_stream: "BOXES:boxes"
+//   input_stream_info: {
+//     tag_index: 'BOXES'
+//     back_edge: true
+//   }
+//   output_stream: "START_POS:start_pos"
+//   output_stream: "CANCEL_ID:cancel_object_id"
+//   output_stream: "ANCHORS:tracked_scaled_anchor_data"
+// }
+
+class TrackedAnchorManagerCalculator : public CalculatorBase {
+private:
+  std::vector<Anchor> previous_anchor_data_;
+  const float box_h_w_ = 0.5; // Used to establish tracking box dimensions
```

Comment:
nit: pull to top of file as a constant.

Also, if I'm understanding the use of this, then I think this may be having us track a very large region of our image.  If that's the case, we should see if something like .1 or .15 helps improve tracking results by keeping us more focused on a smaller patch of the image.